packages/@lwc/template-compiler/src/parser/expression-complex/validate.ts

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import { ParserDiagnostics, invariant } from '@lwc/errors';
+import { walk } from 'estree-walker';
+import * as t from '../../shared/estree';
+import type { BaseNode } from 'estree';
+
+const STATEMENTS_DESC = 'statements are';
```

Comment:
Typically we put these kinds of user-visible English error messages into `error-info/template-transform.ts`.

Reply:
Moved all error language into `@lwc/errors`.